opus_config.cmake

```diff
@@ -0,0 +1,103 @@
+include(opus_functions.cmake)
+
+configure_file(config.h.in config.h @ONLY)
+add_definitions(-DHAVE_CONFIG_H)
+
+option(FIXED_POINT "Use fixed-point code (for devices with less powerful FPU)"
+       OFF)
+option(USE_ALLOCA "Use alloca for stack arrays (on non-C99 compilers)" OFF)
+option(BUILD_PROGRAMS "Build programs" OFF)
+
+set_property(GLOBAL PROPERTY USE_FOLDERS ON)
+set_property(GLOBAL PROPERTY C_STANDARD 99)
+
+if(MSVC)
+  add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+endif()
+
+# It is strongly recommended to uncomment one of these VAR_ARRAYS: Use C99
+# variable-length arrays for stack allocation USE_ALLOCA: Use alloca() for stack
+# allocation If none is defined, then the fallback is a non-threadsafe global
+# array
+if(USE_ALLOCA OR MSVC)
+  add_definitions(-DUSE_ALLOCA)
+else()
+  add_definitions(-DVAR_ARRAYS)
+endif()
+
+include(CheckLibraryExists)
+check_library_exists(m floor "" HAVE_LIBM)
+if(HAVE_LIBM)
+  list(APPEND CMAKE_REQUIRED_LIBRARIES m)
+endif()
+
+opus_supports_cpu_detection(RUNTIME_CPU_CAPABILITY_DETECTION)
+
+include(CMakeDependentOption)
+
+opus_detect_sse(COMPILER_SUPPORT_SIMD)
+if(COMPILER_SUPPORT_SIMD)
+  if(CMAKE_SIZEOF_VOID_P EQUAL 8)
+    set(OPUS_PRESUME_SSE1 ON)
+    set(OPUS_PRESUME_SSE2 ON)
+  else()
+    cmake_dependent_option(OPUS_PRESUME_SSE1
```

Comment:
`OPUS_PRESUME_SSE1` depends on itself?

Reply:
Regarding linker error, can you send the cmake line you are using?

Reply:
Good comments, please have another look.